src/tools/profile/read_profile.c

```diff
@@ -216,54 +371,59 @@ static int show_profile_data_accum(profile_data_t *data, options_t *opts)
             continue;
         }
 
-        loc = &data->locations[sorted_loc->location_idx];
+        loc             = &data->locations[sorted_loc->location_idx];
+        loc_thread_info = &locations_thread_info[sorted_loc->location_idx];
 
         switch (loc->type) {
-        case UCS_PROFILE_TYPE_SAMPLE:
-            printf("%*.*s %13s %13s %10zu %18s:%-4d %s()\n",
-                   FUNC_NAME_MAX_LEN, FUNC_NAME_MAX_LEN,
-                   loc->name,
-                   "-",
-                   "-",
-                   sorted_loc->count,
-                   loc->file, loc->line, loc->function);
-            break;
         case UCS_PROFILE_TYPE_SCOPE_END:
-            printf("%*.*s %13.3f %13.0f %10zu %18s:%-4d %s()\n",
-                   FUNC_NAME_MAX_LEN, FUNC_NAME_MAX_LEN,
-                   loc->name,
-                   time_to_usec(data, opts, sorted_loc->total_time) /
-                   sorted_loc->count,
-                   time_to_usec(data, opts, sorted_loc->total_time),
-                   sorted_loc->count,
-                   loc->file, loc->line, loc->function);
+            snprintf(avg_buf,     sizeof(avg_buf) - 1, "%.3f",
+                     time_to_units(data, opts,
+                                   sorted_loc->total_time / sorted_loc->count));
+            snprintf(total_buf,   sizeof(total_buf) - 1, "%.2f",
+                     time_to_units(data, opts, sorted_loc->total_time));
+            snprintf(overall_buf, sizeof(overall_buf) - 1, "%.3f",
+                     sorted_loc->total_time * 100.0 / loc_thread_info->overall_time);
+
+            avg_str     = avg_buf;
+            total_str   = total_buf;
+            overall_str = overall_buf;
             break;
+        case UCS_PROFILE_TYPE_SAMPLE:
         case UCS_PROFILE_TYPE_REQUEST_EVENT:
-            printf("%*.*s %13s %13s %10zu %18s:%-4d %s()\n",
-                   FUNC_NAME_MAX_LEN, FUNC_NAME_MAX_LEN,
-                   loc->name,
-                   "n/a",
-                   "n/a",
-                   sorted_loc->count,
-                   loc->file, loc->line, loc->function);
+            avg_str = total_str = overall_str = "n/a";
             break;
         default:
-            break;
+            continue;
         }
-    }
+
+        printf("%s%*.*s%s %13s %13s %13s %12zu %s%18s:%-6d %-*s %-13s%s\n",
+               NAME_COLOR, FUNC_NAME_MAX_LEN, FUNC_NAME_MAX_LEN, loc->name, CLEAR_COLOR,
+               avg_str,
+               total_str,
+               overall_str,
+               sorted_loc->count,
+               LOC_COLOR,
+                   loc->file, loc->line,
```

Comment:
broken indent?

Reply:
it's to show that these strings are printed with "LOC_COLOR", which is terminated by "CLEAR_COLOR"